src/gsad_gmp.c

```diff
@@ -4110,6 +4110,240 @@ get_credentials_gmp (gvm_connection_t *connection, credentials_t *credentials,
                    response_data);
 }
 
+/**
+ * @brief Get credential stores, envelope the result.
+ *
+ * @param[in]  connection     Connection to manager.
+ * @param[in]  credentials    Username and password for authentication.
+ * @param[in]  params         Request parameters.
+ * @param[out] response_data  Extra data return for the HTTP response.
+ *
+ * @return Enveloped XML object.
+ */
+char *
+get_credential_stores_gmp (gvm_connection_t *connection,
+                           credentials_t *credentials, params_t *params,
+                           cmd_response_data_t *response_data)
+{
+  const char *credential_store_uuid =
+    params_value (params, "credential_store_id");
+  gmp_arguments_t *arguments = gmp_arguments_new ();
+
+  if (credential_store_uuid)
+    {
+      gmp_arguments_add (arguments, "credential_store_id",
+                         credential_store_uuid);
+    }
+
+  return get_many (connection, "credential_stores", credentials, params,
+                   arguments, response_data);
+}
+
+/**
+ * @brief Modify a credential store, envelope the result.
+ *
+ * @param[in]  connection     Connection to manager.
+ * @param[in]  credentials    Username and password for authentication.
+ * @param[in]  params         Request parameters.
+ * @param[out] response_data  Extra data return for the HTTP response.
+ *
+ * @return Enveloped XML object.
+ */
+char *
+modify_credential_store_gmp (gvm_connection_t *connection,
+                             credentials_t *credentials, params_t *params,
+                             cmd_response_data_t *response_data)
+{
+  gchar *xml, *response, *format;
+  int ret;
+  entity_t entity;
+  const char *credential_store_id, *active, *host, *path, *comment;
+  GHashTable *preferences;
+  GString *preferences_element;
+
+  credential_store_id = params_value (params, "credential_store_id");
+  active = params_value (params, "active");
+  host = params_value (params, "host");
+  path = params_value (params, "path");
+  preferences = params_values (params, "preferences:");
+  comment = params_value (params, "comment");
+
+  if (!credential_store_id)
+    {
+      cmd_response_data_set_status_code (response_data, MHD_HTTP_BAD_REQUEST);
+      return gsad_message (
+        credentials, "Missing credential store id", __func__, __LINE__,
+        "The 'credential_store_id' parameter is required.", response_data);
+    }
```

Comment:
```suggestion
  CHECK_VARIABLE_INVALID (credential_store_id, "Save Credential Store");
  if (str_equal (credential_store_id, ""))
      return message_invalid (connection, credentials, params, response_data,
                              "Missing credential_store_id", "Save Credential Store");  
```
not sure if the formatting in my suggestion is correct